### ISSUE TITLE
Audio Format Question

### ISSUE THREAD

**Issue by mrmuke** [fuwei007/OpenAIIOSRealtimeAPIDemo#5]
Great demo, super useful. I have a question, why do you use pcmFormatInt32 instead of pcmFormatInt16 to play audio? Thank you so much!

**Comment by mrmuke** [comment]
Only reason I ask is because I'm having a lot of trouble trying to play the raw PCM audio properly and would appreciate any help.

**Comment by mrmuke** [comment]
I figured it out! Does't support pcmFormatInt16, thanks for the great app.

**Comment by mrmuke** [comment]
I have another question - why do you need an audio queue if you can just use schedule buffer? Is there any performance benefits?

**Comment by monsterInDream** [comment]
Yes, you can directly use schedule buffer; the audio queue is not essential. I use an audio queue to isolate an audio data source component, which makes it easy to interrupt audio playback or manipulate audio data (such as manually increasing the audio volume) when needed.

**Comment by mrmuke** [comment]
Oh makes sense, I have another question AudioComponentDescription seems to throw an error for me "throwing -1 from AU: auou/vpio/appl, render err: -1," when set to kAudioUnitSubType_VoiceProcessingIO I'm on an iPhone 15 on latest IOS, even though the code works for me. What device are you testing on?

**Comment by monsterInDream** [comment]
This issue has already been raised here: https://github.com/baresip/baresip/discussions/2513. It appears to be a hardware compatibility issue. If it doesn’t affect your usage, you can temporarily ignore it. I’m using an iPhone 12 with iOS 18.1.

**Comment by monsterInDream** [comment]
By the way, I submitted new code with some changes to the output logs.
